src/controllers/user.ts

```diff
@@ -1,107 +1,76 @@
 import type { Request, Response } from 'express';
 import type { ValidatedRequest } from 'express-joi-validation';
-import { v4 as uuidv4 } from 'uuid';
+import { inject, injectable } from 'inversify';
 
-import type { SuggestRequestQueryModel, UserModel } from '../models/User';
+import type { SuggestRequestQueryModel, UserModel } from '../interfaces/User';
+import { UserService } from '../services/user.service';
 import type { UserRequestSchemaModel } from '../validators/user';
 
-const db: { users: UserModel[] } = {
-  users: [
-    {
-      id: '1',
-      password: '1u23',
-      age: 5,
-      login: 'Sasha',
-      isDeleted: false,
-    },
-    {
-      id: '2',
-      password: '1e23',
-      age: 25,
-      login: 'Senya',
-      isDeleted: false,
-    },
-    {
-      id: '3',
-      password: '1c23',
-      age: 35,
-      login: 'Sonya',
-      isDeleted: false,
-    },
-    {
-      id: '4',
-      password: '12b3',
-      age: 15,
-      login: 'Sergey',
-      isDeleted: false,
-    },
-    {
-      id: '4',
-      password: '123a',
-      age: 25,
-      login: 'Sveta',
-      isDeleted: false,
-    },
-  ],
-};
-
-const create = (req: ValidatedRequest<UserRequestSchemaModel>, res: Response) => {
-  const { login, age, password }: UserModel = req.body;
-
-  if (!(login && password && age)) {
-    res.status(400).json({ message: 'Please provide all required fields' });
-
-    return;
-  }
-
-  const user: UserModel = {
-    id: uuidv4(),
-    login,
-    password,
-    age,
-    isDeleted: false,
+@injectable()
+export class UserController {
+  constructor(@inject(UserService) private userService: UserService) {}
+
+  getAll = async (req: Request, res: Response): Promise<void> => {
+    try {
+      const users = await this.userService.getAllUsers();
+
+      if (users.length) {
+        res.status(200).json({ users });
+      } else {
+        res.status(404).json({ message: 'Users not found' });
```

Comment:
Errors should propagate to Error handling middleware via next(err)

Reply:
Fixed, but I have questions to clarify. Let's discuss on Q&A